modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManagerImpl.java

```diff
@@ -98,7 +99,63 @@ public CompletableFuture<Void> schemaReadyFuture(int version) {
         return catalogManager.catalogReadyFuture(version);
     }
 
-    private static SchemaPlus createSqlSchema(int catalogVersion, CatalogSchemaDescriptor schemaDescriptor) {
+    @Override
+    public IgniteTable table(int schemaVersion, int tableId) {
+        return tableCache.get(tableCacheKey(schemaVersion, tableId), key -> {
+            SchemaPlus rootSchema = schemaCache.get(schemaVersion);
+
+            if (rootSchema != null) {
+                for (String name : rootSchema.getSubSchemaNames()) {
+                    SchemaPlus subSchema = rootSchema.getSubSchema(name);
+
+                    assert subSchema != null : name;
+
+                    IgniteSchema schema = subSchema.unwrap(IgniteSchema.class);
+
+                    assert schema != null : "unknown schema " + subSchema;
+
+                    IgniteTable table = schema.getTable(tableId);
+
+                    if (table != null) {
```

Comment:
Currently IgniteSchema#getTable has assertion such that table cannot be null here. I think this assertion should be reworked or removed.

Reply:
I totally missed that, thanks! 

fixed

Reply:
It seems PlanningContext#schema should be fixed also to return the schema that is set in the session.

Reply:
this method is used during execution phase. Why did you mention PlanningContext? PlanningContext is used during planning (aka preparing, see PrepareService)

Reply:
Sorry, may bad, I thought that `config().getDefaultSchema();` returns PUBLIC schema always.